src/Drops.js

```diff
@@ -44,10 +44,25 @@ const Drops = (props) => {
         const drops = await get(dropStorageKey) || []
         return drops.find((d) => d.public_key === public_key)
     }
-    async function updateDrops() {
+    async function updateDrops(check = false) {
         const drops = (await get(dropStorageKey) || [])
         for (let drop of drops) {
-            drop.walletLink = await getWalletLink(drop.public_key)
+            const { public_key: key } = drop
+            drop.walletLink = await getWalletLink(key)
+            if (!check) {
+                continue
+            }
+            // check drop is valid
+            const { contract } = window
+            let res
+            try {
+                res = await contract.get_key_balance({ key })
+            } catch (e) {
+                console.warn(e)
+            }
+            if (!res || nearToInt(res) < 1) {
+                await removeDrop(key)
```

Comment:
@mattlockyer let's better not remove drops from local storage, but just hide them in the list. So that if anything glitches you don't lose the keys.

Right now it seems like you can end up losing your linkdrop key if network is temporarily down (when `contract.get_key_balance` is called).

Reply:
Good point... I'll switch to an "archive" approach and maybe show a button to show those drops if pressed?

Reply:
I think it's best to check whether drops still exist when loading the list, i.e. do not persist it. Or at least require more definite response than any error to indicate that drop already has been claimed.

Other than that – yeah, I think it makes sense to have a button to show "Used drops" or smth like that.